foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,772 @@
+                              location=location, foamobject=foamObject)
+        fileContent = heading + vectorTemplates[solver][data_name]
+        try:
+            f = open(os.path.join(self.path, self.time, data_name), 'w')
+            f.write(fileContent)
+            f.close()
+        except IOError:
+            error_str = "Can't write file: {}"
+            raise ValueError(error_str.format(os.path.join(self.path,
+                                                           self.time,
+                                                           data_name)))
+        FoamFiles().set_all_cell_data(self.path, self.time,
+                                      data_name,
+                                      values, 'vector')
+
+    def create_dummy_celldata(self, data_name, dimensionset):
+        """Created dummy cell data
+
+        Parameters
+        ----------
+        data_name : str
+            Name of data to be created
+        dimensionset : tuple
+            Data dimensionset
+
+        """
+
+        version = '2.2'
+        nCells = foamface.getCellCount(self.name)
+        values = [0.0 for item in range(nCells)]
+        solver = 'interFoam'
+        foamFile = os.path.join(self.time, data_name)
+        foamClass = 'volScalarField'
+        location = '\"' + os.path.dirname(foamFile) + '\"'
+        foamObject = os.path.basename(foamFile)
+        heading = head.format(version=version, foamclass=foamClass,
+                              location=location, foamobject=foamObject)
+        fileContent = heading + scalarTemplates[solver][data_name]
+        try:
+            f = open(os.path.join(self.path, self.time, data_name), 'w')
+            f.write(fileContent)
+            f.close()
+        except IOError:
+            error_str = "Can't write file: {}"
+            raise ValueError(error_str.format(os.path.join(self.path,
+                                                           self.time,
+                                                           data_name)))
+
+        FoamFiles().set_all_cell_data(self.path, self.time,
+                                      data_name,
+                                      values, 'scalar')
+
+    def update_cell(self, cell):
+        """ Updates the information of a cell.
+
+        Gets the mesh cell identified by the same
+        uuid as the provided cell and updates its information
+        with the one provided with the new cell.
+
+        Parameters
+        ----------
+        cell : Cell
+            Cell to be updated
+
+        Raises
+        ------
+        KeyError
+            If the cell was not found in the mesh
+
+        TypeError
+            If the object provided is not a cell
+
+        """
+
+        if cell.uid not in self._uuidToFoamLabel:
+            error_str = "Trying to update a non-existing cell with uuid: "\
+                + str(cell.uid)
+            raise KeyError(error_str)
+
+        if not isinstance(cell, Cell):
```

Comment:
We should avoid checking for the Cell type. Other adapters will not necessarily use the native Cell implementation so the `isinstance(cell, Cell)` will return false even though the object passed supports the Cell interface.